File: packages/modules/api/python/src/api/crossref_dto.py
from dataclasses import dataclass


@dataclass
class Author:
    given: str
    family: str


@dataclass
class Reference:
    id: str
    authors: list[Author]
    title: str
    journal: str
    volume: str | None
    issue: str | None
    pages: str | None
    year: int
    url: str

File: packages/modules/api/typescript/src/crossref.org/crossref.dto.ts
export type Author = { given: string; family: string };

export type CrossReferenceResponse = {
  status: string;
  message: {
    author: { given: string; family: string }[];
    title: string[];
    'container-title': string[];
    volume: string;
    issue: string;
    page: string;
    'published-online': { 'date-parts': number[][] };
    'published-print': { 'date-parts': number[][] };
  };
};

export type Reference = {
  id: string;
  authors: Author[];
  title: string;
  journal: string;
  volume: string;
  issue: string;
  pages: string;
  year: number;
  url: string;
};
